beacon-chain/blockchain/testing/mock_engine.go

```diff
@@ -0,0 +1,57 @@
+package testing
```

Comment:
While we are touching on this topic, could we use the mock client defined in `beacon-chain/powchain/engine-API-client/v1/mocks/client.go` ? I don't see why there should be two mocks, and it's better to use the one next to the implementation. Sorry that `v1/mocks/client.go` only got introduced after `blockchain/mock_engine_test.go`